File: sample-videochat-webrtc/sample-videochat-webrtc/ViewControllers/Settings/SettingSliderCell.m
//
//  SettingSliderCell.m
//  sample-videochat-webrtc
//
//  Created by Andrey Ivanov on 30/09/15.
//  Copyright © 2015 QuickBlox Team. All rights reserved.
//

#import "SettingSliderCell.h"
#import "SliderItemModel.h"

@interface SettingSliderCell()

@property (weak, nonatomic) IBOutlet UILabel *maxLabel;
@property (weak, nonatomic) IBOutlet UILabel *minLabel;
@property (weak, nonatomic) IBOutlet UISlider *slider;

@end

@implementation SettingSliderCell

@synthesize model = _model;

- (void)setModel:(SliderItemModel *)model {
    
    _model = model;
    
    self.label.text = [NSString stringWithFormat:@"%tu", model.currentValue];
    self.maxLabel.text = [NSString stringWithFormat:@"%tu", model.maxValue];
    self.minLabel.text = [NSString stringWithFormat:@"%tu", model.minValue];
    self.slider.minimumValue = model.minValue;
    self.slider.maximumValue = model.maxValue;
    self.slider.value = model.currentValue;
    
    BOOL isEnabled = !model.isDisabled;
    self.slider.enabled = isEnabled;
    self.maxLabel.enabled = isEnabled;
    self.minLabel.enabled = isEnabled;
}

- (IBAction)valueChanged:(UISlider *)sender {
    
    SliderItemModel *model = (SliderItemModel *)self.model;
    model.currentValue = sender.value;
    self.label.text = [NSString stringWithFormat:@"%tu", model.currentValue];
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Settings/SettingSliderCell.swift
//
//  SettingSliderCell.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 12/11/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

class SettingSliderCell: BaseSettingsCell {
    //MARK: - IBOutlets
    @IBOutlet private weak var maxLabel: UILabel!
    @IBOutlet private weak var minLabel: UILabel!
    @IBOutlet private weak var slider: UISlider!
    
    //MARK: - Properties
    var modelSlider = SliderItemModel()
    
    //MARK: - Overrides
    override func updateModel(_ model: BaseItemModel) {
        guard let model = model as? SliderItemModel else { return }
        modelSlider = model
        label.text = "\(modelSlider.currentValue)"
        maxLabel.text = "\(modelSlider.maxValue)"
        minLabel.text = "\(modelSlider.minValue)"
        
        slider.minimumValue = Float(modelSlider.minValue)
        slider.maximumValue = Float(modelSlider.maxValue)
        slider.value = Float(modelSlider.currentValue)
        
        let isEnabled = !modelSlider.isDisabled
        slider.isEnabled = isEnabled
        maxLabel.isEnabled = isEnabled
        minLabel.isEnabled = isEnabled
    }
    
    //MARK: - Actions
    @IBAction func valueChanged(_ sender: UISlider) {
        modelSlider.currentValue = UInt(sender.value)
        label.text = "\(UInt(sender.value))"
    }
}
